packages/hardhat-core/test/internal/hardhat-network/provider/node.ts

```diff
@@ -650,15 +654,15 @@ describe("HardhatNode", () => {
 
         const modifiedBlock = afterBlockEvent.block;
 
-        await forkedNode["_vm"].blockchain.putBlock(modifiedBlock);
-        await forkedNode["_saveBlockAsSuccessfullyRun"](
-          modifiedBlock,
-          afterBlockEvent
-        );
+        // Restore the receipt trie
+        (block as any).header.receiptTrie = modifiedBlock.header.receiptTrie;
 
-        const newBlock = await forkedNode.getBlockByNumber(
-          new BN(blockNumber + 1)
-        );
+        // TODO we should use modifiedBlock instead of block here,
+        // but we can't because of a bug in the vm
```

Comment:
```suggestion
        // but we can't because of a bug in the vm
        // TODO: Change this once https://github.com/ethereumjs/ethereumjs-monorepo/pull/1185 is released.
```